src/Mocking.jl

```diff
@@ -178,26 +113,27 @@ function apply(body::Function, patch::Patch; debug::Bool=false)
     apply(body, PatchEnv(patch, debug))
 end
 
-function ismocked(pe::PatchEnv, func_name::Symbol, args::Tuple)
-    if isdefined(pe.mod, func_name)
-        func = Core.eval(pe.mod, func_name)
-        types = map(arg -> isa(arg, Type) ? Type{arg} : typeof(arg), args)
-        exists = hasmethod(func, types)
+function get_alternate(pe::PatchEnv, target::Callable, args...)
+    if haskey(pe.mapping, target)
+        m, f = dispatch(pe.mapping[target], args...)
+
+        if pe.debug
+            @info "calling mocked method: $m"
+        end
 
+        return f
+    else
         if pe.debug
-            @info("calling $func_name$(types)")
-            if exists
-                m = first(methods(func, types))
-                @info("executing mocked function: $m")
-            else
-                m = first(methods(Core.eval(func_name), types))
-                @info("executing original function: $m")
-            end
+            m, f = dispatch([target], args...)
+            @info "calling original method: $m"
         end
 
-        return exists
+        return nothing
     end
-    return false
+end
+
+function get_alternate(target::Callable, args...)
+    get_alternate(get_active_env(), target, args...)
```

Comment:
```suggestion
    return get_alternate(get_active_env(), target, args...)
```

Reply:
I went with the short-form function syntax